Scripts/Runtime/Entities/Tasks/AbstractTaskDriver.cs

```diff
@@ -90,6 +90,10 @@ protected AbstractTaskDriver(World world, AbstractTaskDriverSystem system)
         {
             World = world;
             System = system;
+            
+            //Generate a Unique ID for this TaskDriver relative to its System
+            m_Context = System.GetNextID();
```

Comment:
Since this value is getting used a whole lot more now I wonder if we should have an implementation of `IDProvider` that uses a smaller value type. A `uint` seems excessive....or are these IDs used for objects other than task drivers?

Implementing a TinyIDProvider would be trivial...maybe even just making IDProvider generic...

Reply:
Agreed. 

PR here: https://github.com/decline-cookies/anvil-csharp-core/pull/109

And I've updated my code to use the Byte one.